packages/react-composites/src/composites/ChatComposite/adapter/FileSharingAdapter.ts

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import produce from 'immer';
+import { FileMetadata, FileSharingMetadata, ObservableFileUpload, FileUploadState } from '../file-sharing';
+import { ChatContext } from './AzureCommunicationChatAdapter';
+import { ChatAdapterState } from './ChatAdapter';
+
+/**
+ * A record containing {@link FileUploadState} mapped to unique ids.
+ * @beta
+ */
+export type FileSharingUiState = Record<string, FileUploadState>;
+
+/**
+ * @beta
+ */
+export interface FileUploadAdapter {
+  registerFileUploads?: (fileUploads: ObservableFileUpload[]) => void;
+  clearFileUploads?: () => void;
+  cancelFileUpload?: (id: string) => void;
```

Comment:
I'm guessing this will be called from the UI when user clicks on a "cancel upload" button?

Reply:
Yes.